addons/mail/static/tests/chatter/web/attachment_box.test.js

```diff
@@ -207,3 +210,53 @@ test("attachment box auto-closed on switch to record wih no attachments", async
     await click(".o_pager_next");
     await contains(".o-mail-AttachmentBox", { count: 0 });
 });
+
+test("Chatter main attachment: can change from non-viewable to viewable", async () => {
+    const pyEnv = await startServer();
+    const resPartnerId = pyEnv['res.partner'].create({});
+    const irAttachmentId = pyEnv['ir.attachment'].create({
+        mimetype: 'text/plain',
+        name: "Blah.txt",
+        res_id: resPartnerId,
+        res_model: 'res.partner',
+    });
+    pyEnv['mail.message'].create({
+        attachment_ids: [irAttachmentId],
+        model: 'res.partner',
+        res_id: resPartnerId,
+    });
+    pyEnv['res.partner'].write([resPartnerId], {message_main_attachment_id : irAttachmentId})
+    const views = {
+        'res.partner,false,form':
+            '<form string="Partners">' +
+                '<sheet>' +
+                    '<field name="name"/>' +
+                '</sheet>' +
+                '<div class="o_attachment_preview"/>' +
+                '<div class="oe_chatter">' +
+                    '<field name="message_ids"/>' +
+                '</div>' +
+            '</form>',
+    };
+    patchUiSize({ size: SIZES.XXL });
+    const { openFormView } = await start({
+        mockRPC(route, args) {
+            if (String(route).includes("/web/static/lib/pdfjs/web/viewer.html")) {
+                var canvas = document.createElement('canvas');
+                return canvas.toDataURL();
+            }
+        },
+        serverData: { views },
+    });
+    await openFormView('res.partner', resPartnerId);
```

Comment:
- `mockRPC` is no longer necessary since 17.2
- `openFormView` is global import rather than returned by `start`
- Should use `registerArchs` to put the `views` rather than passed as `serverData: {.views }` argument to `start`